client/state/posts/selectors.js

```diff
@@ -362,6 +359,61 @@ export const isEditedPostDirty = createSelector(
 );
 
 /**
+ * Returns true if the post status is publish, private, or future
+ * and the date is in the past
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Number}  postId Post ID
+ * @return {Boolean}        Whether post is published
+ */
+export const isPostPublished = createSelector(
+	( state, siteId, postId ) => {
+		const post = getSitePost( state, siteId, postId );
+
+		if ( ! post ) {
+			return null;
+		}
+
+		return includes( [ 'publish', 'private' ], post.status ) ||
+			( post.status === 'future' && moment( post.date ).isBefore( moment() ) );
+	},
+	( state ) => state.posts.items
```

Comment:
`state.posts.queries` should be the cachebust here. `items` is just a mapping and doesn't necessarily reflect changes in the underlying objects.